src/middleware/auth/authanticated.middleware.ts

```diff
@@ -0,0 +1,90 @@
+import { NextFunction, Request, Response } from "express";
+import { verifyToken } from "../../utils";
+import { isValidUuid } from "../../utils";
+
+export const AuthAndRoleChecker = (key: () => { value: string }) => {
+  return async (req: Request, res: Response, next: NextFunction) => {
+    if (!req.headers.authorization) {
+      res.status(401).send({
+        status: 401,
+        success: false,
+        message: "Unauthorized",
+      });
+    } else {
+      if (isValidUuid(key().value)) {
+        const accessKey = key().value;
+        try {
+          const token = req.headers.authorization.split(" ")[1];
+          const {
+            id,
```

Comment:
the same applies here, you're verifying this encoded data but you shouldn't encode this much data in the first place